xdsl/dialects/riscv.py

```diff
@@ -524,26 +514,9 @@ def _assembly_arg_str(arg: AssemblyInstructionArg) -> str:
     assert_never(arg)
 
 
-def _assembly_line(
-    name: str,
-    arg_str: str,
-    comment: StringAttr | None = None,
-    is_indented: bool = True,
-) -> str:
-    code = "    " if is_indented else ""
-    code += name
-    if arg_str:
-        code += f" {arg_str}"
-    code = _append_comment(code, comment)
-    return code
-
-
 def print_assembly(module: ModuleOp, output: IO[str]) -> None:
-    for op in module.body.walk():
-        assert isinstance(op, RISCVAsmOperation), f"{op}"
-        asm = op.assembly_line()
-        if asm is not None:
-            print(asm, file=output)
+    printer = AssemblyPrinter(stream=output)
+    printer.print_module(module)
```

Comment:
I wanted to keep this around not just to minimise diff but also to allow for future RISCVAssemblyPrinter or whatever if we want to add some special helper functionality.